### PR TITLE
Extract util/errors.ts into a new common package

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
As part of the process of Extracting non-UI modules from the shared package into a new workspaces, extract the `util/errors.ts` module into a new common package

## Refs
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/21847)
[Gitstart Issue](https://app.gitstart.com/clients/sourcegraph/tickets/SG-28892)

## Time estimate

- Pull requests with ~450 lines changed should take **3 hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.
- Split the work into multiple pull requests if the total diff is bigger than 450 lines of code.

### CODE REVIEW

File: client/common/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+### This package contains common files
```
Comment (valerybugakov) [inline client/common/README.md:1]:
```suggestion
# Common utilities

Generic utilities without any specific domain.
This package should not depend on any other Sourcegraph monorepo package.
```

File: client/common/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// @ts-check
+
+const config = require('../../jest.config.base')
+
+const exportedConfig = {
+  ...config,
+  displayName: 'common',
+  rootDir: __dirname,
+  collectCoverage: false, // Collected through Puppeteer
```
Comment (valerybugakov) [inline client/common/jest.config.js:9]: Oh, this should not be here. Let's remove this line from the `build-config` package too.

```suggestion
```

File: client/common/package.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "private": true,
+  "name": "@sourcegraph/common",
+  "version": "0.0.1",
+  "description": "Sourcegraph Common Files",
```
Comment (valerybugakov) [inline client/common/package.json:5]:
```suggestion
  "description": "Sourcegraph common utilities",
```

File: client/shared/tsconfig.json [hunk 4]
```diff
@@ -11,6 +11,6 @@
     "rootDir": ".",
     "outDir": "./out",
   },
-  "include": ["**/*", ".*", "./src/**/*.json"],
+  "include": ["**/*", ".*", "./src/**/*.json", "/client/common/**/*.ts"],
```
Comment (valerybugakov) [inline client/shared/tsconfig.json:14]:
```suggestion
  "include": ["**/*", ".*", "./src/**/*.json"],
```

File: client/shared/tsconfig.json [hunk 5]
```diff
@@ -11,6 +11,6 @@
...
   "exclude": ["../../node_modules", "./node_modules", "./out", "./src/end-to-end"],
```
Comment (valerybugakov) [inline client/shared/tsconfig.json:15]: A reference should be added to every package that uses `@sourcegraph/common`. Would you please double-check other packages and update `tsconfig.json` accordingly?

```suggestion
  "exclude": ["../../node_modules", "./node_modules", "./out", "./src/end-to-end"],
  "references": [{ "path": "../common" }],
```

File: tsconfig.all.json [hunk 6]
```diff
@@ -2,6 +2,7 @@
   "extends": "./tsconfig.json",
   "references": [
     { "path": "client/build-config" },
+    { "path": "client/common" },
```
Comment (valerybugakov) [inline tsconfig.all.json:5]: `.github/workflows/lsif-ts.yml` should be updated to include a new package.

**Review by valerybugakov — CHANGES_REQUESTED** [review]

**Comment by gitstart-sourcegraph** [comment]
Hi @valerybugakov  updated with requested changes

File: client/common/src/errors/types.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { AGGREGATE_ERROR_NAME } from './constants'
+export interface ErrorLike {
```
Comment (valerybugakov) [inline client/common/src/errors/types.ts:2]:
```suggestion
import { AGGREGATE_ERROR_NAME } from './constants'

export interface ErrorLike {
```

File: yarn.lock [hunk 8]
```diff
@@ -6430,7 +6430,7 @@ any-promise@^1.0.0, any-promise@^1.1.0, any-promise@~1.3.0:
   resolved "https://registry.npmjs.org/any-promise/-/any-promise-1.3.0.tgz#abc6afeedcea52e809cdc0376aed3ce39635d17f"
   integrity sha1-q8av7tzqUugJzcA3au0845Y10X8=
 
-anymatch@*, anymatch@^3.0.3, anymatch@~3.1.1, anymatch@~3.1.2:
+anymatch@*, anymatch@^3.0.3, anymatch@~3.1.1:
```
Comment (valerybugakov) [inline yarn.lock:6433]: Let's merge the `main` branch into this PR to ensure that this redundant change goes away. It shouldn't be present in the diff.

**Review by valerybugakov — CHANGES_REQUESTED** [review]
I see that Percy failed with the issue already addressed in the `main` branch. Let's merge the `main` branch to ensure that CI checks are green.

**Review by valerybugakov — APPROVED** [review]
Thanks!

**Pull request merged**